test/local/dataflow_tests.js

```diff
@@ -0,0 +1,162 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+const proxyquire = require('proxyquire')
+const sinon = require('sinon')
+const { test } = require('tap')
+
+let log
+let config
+let dataflow
+let PubSub
+let pubsub
+let subscription
+
+test(
+  'shared setup', async (t) => {
```

Comment:
How come `async`? I don't see any `await` or promise-related stuff here or in the module under test...

Reply:
Because of tap. If the function returns a promise, then tap does not require you to call t.end. However if you do not return a promise, you have to call t.end. It seemed cleaner this way.

Reply:
Ah, makes sense, thanks.